### PR TITLE
Fix Firestore conformance tests

### LINKED ISSUE #5343: Some Firestore conformance tests are currently ignored
We need to fix the conformance tests, then unskip them.

### PR DESCRIPTION
**Pull request by jskeet**
Fixes #5343.

This should not be merged until we've seen a spec for update transforms.

### CODE REVIEW

**Review by amanda-tarafa — COMMENTED** [review]
LGTM.
I've added the `do not merge` label and won't really approve so we don't accidentally merge before we have confirmation.

**Comment by jskeet** [comment]
Note: using the precondition to determine whether or not to use an empty update mask may not be a good idea. It *may* be as simple as depending on which method is calling it (so just have an extra flag...)

**Comment by jskeet** [comment]
Amanda: I've added an extra commit to this based on internal guidance. Please could you have a look? If you're happy, we can merge this and we're done :)

File: apis/Google.Cloud.Firestore/Google.Cloud.Firestore/WriteBatch.cs [hunk 1]
```diff
@@ -264,7 +244,9 @@ internal async Task<IList<WriteResult>> CommitAsync(ByteString transactionId, Ca
                     Fields = { fields },
                     Name = documentReference.Path,
                 },
-                UpdateMask = includeEmptyUpdateMask || updatePaths.Count > 0 ? new DocumentMask { FieldPaths = { updatePaths.Select(fp => fp.EncodedPath) } } : null,
+                UpdateMask = updatePaths is null
```
Comment (jskeet) [inline apis/Google.Cloud.Firestore/Google.Cloud.Firestore/WriteBatch.cs:247]: I could compute the DocumentMask (or null) outside the object initializer if you'd prefer.

Comment (amanda-tarafa) [reply apis/Google.Cloud.Firestore/Google.Cloud.Firestore/WriteBatch.cs:247]: No, this is fine.

**Review by amanda-tarafa — APPROVED** [review]
Yep, so much cleaner.

**Pull request merged**